src/controllers/redirect.controller.ts

```diff
@@ -53,21 +53,29 @@ export class RedirectController extends BaseController {
       case ApplicationStatus.PENDING_PAYMENT:
         return this.handlePendingPaymentRedirect(isApplicant)
       case ApplicationStatus.PAID:
-        return this.redirect(CONFIRMATION_URI)
+
+        return this.redirect(VIEW_FINAL_CONFIRMATION_URI)
     }
   }
 
   @httpGet('/payment-callback')
   public async getPaymentCallback(@queryParam('state') state: string,
-                                  @queryParam('status') status: string): Promise<RedirectResult> {
-    const dissolutionSession: DissolutionSession = this.session.getDissolutionSession(this.httpContext.request)!
-
+                                  @queryParam('status') status: string,
+                                  @queryParam('ref') reference: string): Promise<RedirectResult> {
+    const dissolutionSession: DissolutionSession = this.updateSessionWithReferenceNumber(reference)
     if (status === PaymentStatus.PAID && dissolutionSession.paymentStateUUID === state) {
-      return this.redirect(CONFIRMATION_URI)
+      return this.redirect(VIEW_FINAL_CONFIRMATION_URI)
     }
     return this.redirect(ERROR_URI)
   }
 
+  private updateSessionWithReferenceNumber(reference: string): DissolutionSession {
+    const dissolutionSession: DissolutionSession = this.session.getDissolutionSession(this.httpContext.request)!
```

Comment:
we already have access to dissolutionSession in the `/redirect` method, so let's only fetch session once per public method

I also think it would be good if we replaced the redirects with a private method called `saveSessionAndRedirect`:
```
public async get(): Promise<RedirectResult> {
  const dissolutionSession: DissolutionSession = this.session.getDissolutionSession(this.httpContext.request)!
  dissolutionSession. applicationReferenceNumber = dissolution.application_reference

  switch:
  ....
  return this.saveSessionAndRedirect(session, VIEW_FINAL_CONFIRMATION_URI)
}

public async getPaymentCallback(...) {
  const session: DissolutionSession = this.session.getDissolutionSession(this.httpContext.request)!
  session. applicationReferenceNumber = ref

  if (status === PaymentStatus.PAID && dissolutionSession.paymentStateUUID === state) {
    return this.saveSessionAndRedirect(session, VIEW_FINAL_CONFIRMATION_URI)
  }

  return this.saveSessionAndRedirect(session, ERROR_URI)
}

private saveSessionAndRedirect(session: DissolutionSession, redirectUri: string): RedirectResult {
  this.session.setDissolutionSession(this.httpContext.request, dissolutionSession)
  return this.redirect(redirectUri)
}
```